File: 79-word-search/word-search.c
bool exist(char** board, int boardSize, int* boardColSize, char* word) {
    
    
 int m = boardSize;
    int n = boardColSize[0];
    int len = strlen(word);

   
    bool dfs(int i, int j, int index) {
        if (index == len) return true;
        if (i < 0 || i >= m || j < 0 || j >= n) return false;
        if (board[i][j] != word[index]) return false;

        char temp = board[i][j];
        board[i][j] = '#';  

        bool found = dfs(i + 1, j, index + 1) ||
                     dfs(i - 1, j, index + 1) ||
                     dfs(i, j + 1, index + 1) ||
                     dfs(i, j - 1, index + 1);

        board[i][j] = temp;  
        return found;
    }

    
    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            if (dfs(i, j, 0)) return true;
        }
    }

    return false;
}   

File: 79-word-search/word-search.py
class Solution(object):
    def exist(self, board, word):
        """
        :type board: List[List[str]]
        :type word: str
        :rtype: bool
        """
        m, n = len(board), len(board[0])
        
        def dfs(i, j, index):
            if index == len(word):
                return True
            if i < 0 or i >= m or j < 0 or j >= n or board[i][j] != word[index]:
                return False
            
            temp = board[i][j]
            board[i][j] = '#' 
            
            found = (dfs(i + 1, j, index + 1) or
                     dfs(i - 1, j, index + 1) or
                     dfs(i, j + 1, index + 1) or
                     dfs(i, j - 1, index + 1))
            
            board[i][j] = temp  
            return found
        
        for i in range(m):
            for j in range(n):
                if dfs(i, j, 0):
                    return True
        return False